<a href="https://colab.research.google.com/github/zmhibner-gif/ml_internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/zmhibner-gif/ml_internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [ ]:
# Set up the repo in Colab
!git clone -q https://github.com/zmhibner-gif/ml_internship.git /content/ml_internship
%cd /content/ml_internship

/content/ml_internship



## 1. Lane 3: Structured Content Archetype Clustering using the starter dataset.

I want to focus on low-performing pages and explore whether their search performance signals form distinct recurring groups. M main aim is to investigate whether differences in metrics such as impressions, CTR, and average position reveal useful archetypes that could support different review or optimization actions. I chose this lane because clustering may help uncover patterns without requiring predefined labels, while still keeping the result exploratory rather than treating the clusters as objectively true page types. I am also alresy familiar with clustering since i studied it in univerisity.

## 2. The question: decision, action, cost of a wrong call

**Research question:** Can low-performing pages be grouped into distinct search performance archetypes using impressions, CTR, and average position, so content teams can choose a more appropriate optimization action?

**Unit of analysis:** one page/content item. **Output:** a cluster assignment and a profile describing each recurring performance pattern.

The decision is how a content or SEO team should respond to an underperforming page. A simple rule such as “zero clicks = poor performance” may group together pages that are underperforming in different ways, while clustering can combine several observed signals to explore recurring patterns. A wrong recommendation could waste time on the wrong pages or lead to the wrong intervention, so the clusters should support investigation rather than make decisions automatically.


## 3. Quick look at the data



In [ ]:
import pandas as pd

# Load the starter dataset from the repository
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Count the total number of pages in the dataset.
# Each row represents one content page.
total_pages = len(df)

# Count how many pages received no clicks in the last 90 days.
# This helps us see how common low click performance is.
zero_clicks = (df["clicks_90d"] == 0).sum()

# Convert the number of zero-click pages into a percentage
# so it is easier to compare with the whole dataset.
zero_clicks_pct = zero_clicks / total_pages * 100

# Look for pages that had some search visibility but still received no clicks.
# I use 1,000 impressions here only as a simple exploratory threshold,
# not as a definition of what "high impressions" means.
high_impressions_zero_clicks = (
    (df["impressions_90d"] >= 1000)
    & (df["clicks_90d"] == 0)
).sum()

# Print the numbers that support the research question.
print(f"Total pages: {total_pages:,}")
print(f"Pages with zero clicks: {zero_clicks:,} ({zero_clicks_pct:.1f}%)")
print(
    f"Pages with at least 1,000 impressions but zero clicks: "
    f"{high_impressions_zero_clicks:,}"
)


Total pages: 30,000
Pages with zero clicks: 13,204 (44.0%)
Pages with at least 1,000 impressions but zero clicks: 1,149


The dataset contains 30,000 pages, and 13,204 pages (44.0%) received zero clicks in the last 90 days. However, 1,149 pages had at least 1,000 impressions and still received zero clicks. This suggests that pages with the same outcome, such as zero clicks, may be underperforming for different reasons. Some may have low visibility, while others are visible in search but do not attract clicks. This supports exploring whether clustering can reveal different search performance archetypes among underperforming pages.


## 4. Careful words: what I can and can't claim

This analysis can show whether pages with similar search-performance metrics tend to form recurring groups in the starter dataset. These clusters may help describe different patterns of underperformance and support further investigation.

However, the clusters are not true labels, and they do not prove why a page is underperforming or what action will definitely improve it. I also cannot claim that the analysis explains Google's ranking system. The results should be treated as exploratory patterns that may help with prioritization and decision support.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.